django/core/management/commands/shell.py

```diff
@@ -111,10 +115,35 @@ def python(self, options):
         # Start the interactive interpreter.
         code.interact(local=imported_objects)
 
+    def get_and_report_namespace(self, verbosity):
+        namespace = self.get_namespace()
+
+        if verbosity >= 1:
+            self.stdout.write(
+                f"{len(namespace)} objects imported automatically",
+                self.style.SUCCESS,
+            )
```

Comment:
This is untested :+1:

Reply:
Right now I'm not able to test this 😥. This afternoon I'll do it.

Reply:
Thank you!
We could tweak this to list a couple if we want but don't need to :+1:

Reply:
What do you mean by "list a couple"?

Reply:
Like: `7 objects imported automatically, including: Question, Choice and User` but it might over complicate this logic

Reply:
I don’t think this would be helpful, as it would be arbitrary.

Reply:
https://forum.djangoproject.com/t/default-automatic-imports-in-the-shell/33708/6
Listing everything is more informative than a count - do you think we can add this here?

Reply:
Yes, let’s list everything when `verbosity` is >=2, at least Andy (nanorepublica) and I are in consensus on the forum thread.

It will take a little care @salvo-polizzi - you won't reconstitute import statements for every object, only functions and classes, which have a `__module__` attribute. And iff a `__qualname__` has dots, that represents an inner class or method.

Reply:
@adamchainz How should we display the imports? Should we use a line for each imported object, like 'from a.b import c,' or should we just say 'Imported a.b.c'?

Reply:
One more question: Do we want to list the models' modules as well, or not? (If I understand correctly, we shouldn't.)